src/BTVNanoCommissioning/utils/histogrammer.py

```diff
@@ -0,0 +1,196 @@
+from BTVNanoCommissioning.helpers.definitions import definitions
+import hist as Hist
+
+
+def histogrammer(workflow):
+    _hist_dict = {}
+    ## Common variables
+    flav_axis = Hist.axis.IntCategory([0, 1, 4, 5, 6], name="flav", label="Genflavour")
+    syst_axis = Hist.axis.StrCategory([], name="syst", growth=True)
+    pt_axis = Hist.axis.Regular(50, 0, 300, name="pt", label=" $p_{T}$ [GeV]")
+    softlpt_axis = Hist.axis.Regular(25, 0, 25, name="pt", label=" $p_{T}$ [GeV]")
+    mass_axis = Hist.axis.Regular(50, 0, 300, name="mass", label=" $p_{T}$ [GeV]")
+    eta_axis = Hist.axis.Regular(25, -2.5, 2.5, name="eta", label=" $\eta$")
+    phi_axis = Hist.axis.Regular(30, -3, 3, name="phi", label="$\phi$")
+    mt_axis = Hist.axis.Regular(30, 0, 300, name="mt", label=" $m_{T}$ [GeV]")
+    iso_axis = Hist.axis.Regular(30, 0, 0.15, name="pfRelIso03_all", label="Rel. Iso")
+    softliso_axis = Hist.axis.Regular(
+        20, 0.2, 4.2, name="pfRelIso03_all", label="Rel. Iso"
+    )
+    dr_axis = Hist.axis.Regular(20, 0, 4, name="dr", label="$\Delta$R")
+    sliso_axis = Hist.axis.Regular(
+        30, 0.2, 4.0, name="pfRelIso03_all", label="Rel. Iso"
+    )
+    dxy_axis = Hist.axis.Regular(40, -0.05, 0.05, name="dxy", label="d_{xy}")
+    dz_axis = Hist.axis.Regular(40, 0, 0.1, name="dz", label="d_{z}")
+    sip3d_axis = Hist.axis.Regular(20, 0, 0.2, name="sip3d", label="SIP 3D")
+    ptratio_axis = Hist.axis.Regular(50, 0, 1, name="ratio", label="ratio")
+    n_axis = Hist.axis.IntCategory([0, 1, 2, 3, 4, 5], name="n", label="N obj")
+
+    ### Workflow specific
+    if "ttdilep_sf" == workflow:
+        obj_list = ["mu", "ele"]
+        for i in range(2):
+            obj_list.append(f"jet{i}")
+            _hist_dict[f"dr_mujet{i}"] = Hist.Hist(
+                flav_axis, dr_axis, Hist.storage.Weight()
+            )
+        for i in ["mu", "ele"]:
+            _hist_dict[f"{i}_pfRelIso04_all"] = Hist.Hist(
+                iso_axis, Hist.storage.Weight()
+            )
+            _hist_dict[f"{i}_dxy"] = Hist.Hist(dxy_axis, Hist.storage.Weight())
+            _hist_dict[f"{i}_dz"] = Hist.Hist(dz_axis, Hist.storage.Weight())
+    if "ttsemilep_sf" == workflow:
+        obj_list = ["mu", "MET"]
+        for i in range(4):
+            obj_list.append(f"jet{i}")
+            _hist_dict[f"dr_mujet{i}"] = Hist.Hist(
+                flav_axis, dr_axis, Hist.storage.Weight()
+            )
+        for i in ["mu"]:
```

Comment:
semileptonic workflow is only for mu+jets?

Reply:
The jet kinematics and discriminator score are stored in [obj_list](https://github.com/cms-btv-pog/BTVNanoCommissioning/pull/30/files/72a7d0e041e972a05455f6e4cffe23b8d7bd36fb#diff-687f01ad2691aae1d5e03b2a5d42bf28a9b7882284a0969dc449d29bd0a9c0ccR47)